internal/lib/lib.go

```diff
@@ -711,17 +719,33 @@ func InformersToRegister(oclient *oshiftclient.Clientset, kclient *kubernetes.Cl
 	return allInformers, nil
 }
 
-func SSLKeyCertChecksum(sslName, certificate, cacert string) uint32 {
-	return utils.Hash(sslName + certificate + cacert)
+func SSLKeyCertChecksum(sslName, certificate, cacert string, markers []*models.RoleFilterMatchLabel, populateCache bool) uint32 {
+	checksum := utils.Hash(sslName + certificate + cacert)
+	if GetGRBACSupport() {
+		if markers != nil && populateCache {
+			checksum += ObjectLabelChecksum(markers)
+		} else {
+			checksum += GetClusterLabelChecksum()
+		}
+	}
+	return checksum
 }
 
-func L4PolicyChecksum(ports []int64, protocol string) uint32 {
+func L4PolicyChecksum(ports []int64, protocol string, markers []*models.RoleFilterMatchLabel, populateCache bool) uint32 {
 	var portsInt []int
 	for _, port := range ports {
 		portsInt = append(portsInt, int(port))
 	}
 	sort.Ints(portsInt)
-	return utils.Hash(utils.Stringify(portsInt)) + utils.Hash(protocol)
+	checksum := utils.Hash(utils.Stringify(portsInt)) + utils.Hash(protocol)
+	if GetGRBACSupport() {
+		if markers != nil && populateCache {
```

Comment:
@chauhanshubham , @sudswasavi : I feel this logic still have an issue. Lets say customer with older avi controller (having DS, sslkey objects etc) migrates to 20.1.5. So when AKO fetches objects (L3 side) from AVI controller, for DS objects say, Markers field will not be there. So it will be marked nil. So we will go to Else part and add labels value to checksum. Now ate L2 side, while computing we will also use same cksum logic. So cache obj cksum and computed cksum will be same, so we will not do PUT call to add markers to these objects. So these object will remain without Markers(labels).